components/ui/datepicker/component.js

```diff
@@ -60,6 +61,23 @@ const Datepicker = ({ lang, selected, minDate, maxDate, ...props }) => {
     maxDate.getTime() + maxDate.getTimezoneOffset() * 60000
   );
 
+  const selectedMomentDate = moment(selectedUTC);
+  const minMomentDate = moment(minDateUTC);
+  const maxMomentDate = moment(maxDateUTC);
+
+  const diffFromStart = minMomentDate.diff(selectedMomentDate, 'days');
+  const diffFromEnd = maxMomentDate.diff(selectedMomentDate, 'days');
+
+  if (diffFromStart > 0) {
```

Comment:
you don't need two ifs for this logic.

```suggestion
selectedUTC = minDateUTC;

  if (diffFromStart < 0) {
      selectedUTC = maxDateUTC;
  }
```

Reply:
in this case `selectedUTC` already has a default value, which is 
```
new Date(
    selected.getTime() + selected.getTimezoneOffset() * 60000
  );
```

which is different than the if cases

hence we can't leave it with 1 if since we need to check both diffs